dataformat-xml-dom/src/main/java/org/camunda/spin/impl/xml/dom/format/DomXmlDataFormat.java

```diff
@@ -53,6 +53,8 @@
 
   protected final String name;
 
+  protected boolean prettyPrint = true;
```

Comment:
Setting the `prettyPrint` property to true would break backward compatibility. Any users that currently expect the output to have an empty line would face unexpected behaviors.

We should also keep with the pattern of setting property values in the constructor.

I would remove the value here:
```suggestion
  protected boolean prettyPrint;
```

And adjust the constructor on lines 75-80 to set the `prettyPrint` property. By default, `false` should be passed.

Reply:
in the current master branch DomXmlDataFormatWriter#getTransformer returns a transformer with property INDENT set - so with default setting it DOES formatting. With prettyPrint = false we have added the capability to do no transformation and write the XML as is into the variable. But this is not today's behavior - that is why prettyPrint defaults to true. 

The root cause is, that the transformer with property INDENT set only works as expected, if the input XML is not already pretty printed. I think this can even depend on the implementation of javax.xml.Transform on the classpath which is is used when transformerFactory.newTransformer(); is called.  So probably very hard to fix if at all. 

The formatter behavior is as follows: 
- If you put in an unformatted XML (everything in one line), the output is a formatted XML as expected
- if you put in an already formatted XML, the formatter adds an additional newline behind each line leading to blank lines. 
- if you put this output another time into the formatter you have already 2 blank lines after each line
- if the property INDENT is not set (prettyPrint = false) the output will not be touched